### PR TITLE
feat: optimize material group

### PR DESCRIPTION
**Pull request by hexqi**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### 物料分组相关优化
- 组件物料分组显示中文，添加label字段
- 调整新分组与TinyVue组件分组一致：布局与容器、基础元素、高级元素、表单、表格、导航、数据展示、ElementPlus
- 删除与TinyVue组件重复的HTML组件及HTML分组：button、img、table、form，部分HTML移入基础元素
- TinyRow与TinyCol合并为 栅格布局组件
- 添加物料时支持合并snippets分组
- 物料分组不区分 内置组件、HTML组件
- 其他：部分代码优化、规范物料文件格式

### 之前分组截图
image
image

### 新分组截图
image
image

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Introduced new UI components including `TinySearch` and `TinyDatePicker`.
	- Enhanced internationalization support for component rendering based on user locale.

- **Improvements**
	- Renamed and reorganized existing components for clearer categorization and better usability.
	- Streamlined component registration process, improving clarity and organization.

- **Bug Fixes**
	- Improved handling of event schemas for built-in components.

- **Documentation**
	- Updated documentation to reflect new component names and structures for easier developer reference.

### CODE REVIEW

File: packages/plugins/materials/src/composable/useMaterial.js [hunk 1]
```diff
@@ -248,15 +248,51 @@ const setThirdPartyDeps = (components) => {
...
+const addComponentSnippets = (componentSnippets, snippetsData) => {
+  if (!componentSnippets) return
+
+  const snippetsMap = new Map()
+  snippetsData.forEach((snippetGroup) => snippetsMap.set(snippetGroup.group, snippetGroup))
+  componentSnippets.forEach((snippetGroup) => {
+    if (snippetsMap.has(snippetGroup.group)) {
+      snippetsMap.get(snippetGroup.group).children.push(...snippetGroup.children)
+    } else {
+      snippetsData.push(snippetGroup)
```
Comment (chilingling) [inline packages/plugins/materials/src/composable/useMaterial.js:265]: `componentSnippets` 有没有可能有同名的 `group` ?

File: packages/plugins/materials/src/meta/component/src/Main.vue [hunk 2]
```diff
@@ -4,7 +4,12 @@
       <template #prefix> <icon-search /> </template>
     </tiny-search>
     <tiny-collapse v-model="state.activeName" class="lowcode-scrollbar">
-      <tiny-collapse-item v-for="(item, index) in state.components" :key="item.group" :title="item.group" :name="index">
+      <tiny-collapse-item
+        v-for="(item, index) in state.components"
+        :key="item.group"
+        :title="item.label?.zh_CN || item.group"
```
Comment (chilingling) [inline packages/plugins/materials/src/meta/component/src/Main.vue]: 这里是不是可以直接 `item.label[locale]` （直接支持国际化）

File: packages/builtinComponent/src/meta/index.js [hunk 3]
```diff
@@ -3,16 +3,14 @@ import CanvasRow from './CanvasRow.json'
...
-      group: '内置组件',
-      children: [{ ...CanvasCol.components }, { ...CanvasRow.components }, { ...CanvasRowColContainer.components }]
-    }
-  ],
+  components: [CanvasCol.component, CanvasRow.component, CanvasRowColContainer.component],
   snippets: [
     {
-      group: '内置组件',
-      children: [{ ...CanvasRowColContainer.snippets }]
+      group: 'layout',
```
Comment (chilingling) [inline packages/builtinComponent/src/meta/index.js:9]: 这里 `group` 充当的应该是 `id` 的效果，是否可以改成 `id` 或者是 `groupId` ？
另外就是这里相当于改协议了，建议到时文档标注。不然原有的项目物料会有问题。

File: packages/plugins/materials/src/composable/useMaterial.js (hunk 1 above)
Comment (hexqi) [reply packages/plugins/materials/src/composable/useMaterial.js:265]: 物料本身提供了同名的group，属于物料的问题，我觉得不用考虑

File: packages/plugins/materials/src/meta/component/src/Main.vue (hunk 2 above)
Comment (hexqi) [reply packages/plugins/materials/src/meta/component/src/Main.vue]: 我改下。只是我们设计器基本都没国际化，有点尴尬。

File: packages/builtinComponent/src/meta/index.js (hunk 3 above)
Comment (hexqi) [reply packages/builtinComponent/src/meta/index.js:9]: 我理解这样改动不会有影响，反而对用户的影响是最小的：之前拿group当组名用的场景，group是中文，由于没有label字段，显示上还是会用group，仍是中文，不会有影响，之前没有group当id的场景所以不用考虑。
文档这块确实需要改下。

**Review by chilingling — APPROVED** [review]

File: packages/builtinComponent/src/meta/index.js (hunk 3 above)
Comment (hexqi) [reply packages/builtinComponent/src/meta/index.js:9]: 文档已在portal仓库补充

**Review by rhlin — APPROVED** [review]

**Pull request merged**